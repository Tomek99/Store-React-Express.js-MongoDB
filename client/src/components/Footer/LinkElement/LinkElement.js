import React from "react";
import { Link } from "react-router-dom";
import styles from "./LinkElement.module.scss";

function LinkElement({ item }) {
  const { name, path } = item;
  return (
    <li className={styles.LinkElement}>
      <Link to={path}>{name}</Link>
    </li>
  );
}

export default LinkElement;
