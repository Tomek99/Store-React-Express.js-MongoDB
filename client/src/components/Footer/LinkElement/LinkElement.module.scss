.LinkElement {
    a {
        color: #fff;
        position: relative;

        &::before {
            content: "";
            position: absolute;
            left: 0;
            bottom: -5px;
            width: 0;
            height: 2px;
            background-color: var(--main-color);
            transition: all 0.3s ease;
        }

        &:hover::before {
            width: 100%;
        }
    }
}
