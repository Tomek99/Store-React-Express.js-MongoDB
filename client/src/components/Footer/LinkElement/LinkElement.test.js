import { render, screen } from "@testing-library/react";
import LinkElement from "./LinkElement";
import { MemoryRouter } from "react-router-dom";
test("should render LinkElement component", async () => {
  const item = { name: "test", path: "test" };
  const { container } = render(
    <MemoryRouter>
      <LinkElement item={item} />
    </MemoryRouter>
  );

  await screen.findByRole("link");
  expect(container).toMatchSnapshot();
});
