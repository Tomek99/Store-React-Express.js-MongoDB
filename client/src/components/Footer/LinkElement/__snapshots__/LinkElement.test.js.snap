// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LinkElement component 1`] = `
<div>
  <li
    class="LinkElement"
  >
    <a
      href="/test"
    >
      test
    </a>
  </li>
</div>
`;
